File: app/src/main/java/project/android/headli9es/ConfigActivity.java
package project.android.headli9es;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.preference.ListPreference;
import android.preference.Preference;
import android.preference.PreferenceFragment;
import android.preference.PreferenceManager;

import androidx.appcompat.app.AppCompatActivity;

public class ConfigActivity extends AppCompatActivity {

    @Override
    protected void onCreate (Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_config);
    }

    public static class NewsConfigFragment extends PreferenceFragment
            implements Preference.OnPreferenceChangeListener {

        @Override
        public void onCreate (Bundle savedInstanceState) {
            super.onCreate(savedInstanceState);
            addPreferencesFromResource(R.xml.configs);

            // Find the {@link ListPreference}s in configs.xml
            Preference newsOutlet = findPreference(getString(R.string.settings_news_outlet_key));
            Preference pageSize = findPreference(getString(R.string.settings_page_size_key));

            // Set and display summary of the {@link ListPreference} according to selected value
            setSummary(newsOutlet);
            setSummary(pageSize);
        }

        private void setSummary (Preference changedPreference) {
            changedPreference.setOnPreferenceChangeListener(this);
            SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(changedPreference.getContext());
            String preferenceString = preferences.getString(changedPreference.getKey(), "");
            assert preferenceString != null;
            onPreferenceChange(changedPreference, preferenceString);
        }

        @Override
        public boolean onPreferenceChange (Preference preference, Object newValue) {
            String preferenceValue = newValue.toString();

            if (preference instanceof ListPreference) {
                ListPreference listPref = (ListPreference) preference;
                final int indexOfSelectedValue = listPref.findIndexOfValue(preferenceValue);
                if (indexOfSelectedValue >= 0) {
                    CharSequence[] prefLabels = listPref.getEntries();
                    preference.setSummary(prefLabels[indexOfSelectedValue]);
                }
                return true;
            } else
                return false;
        }
    }
}

File: app/src/main/java/project/android/headli9es/ConfigActivity.kt
package project.android.headli9es

import android.os.Bundle
import android.preference.ListPreference
import android.preference.Preference
import android.preference.Preference.OnPreferenceChangeListener
import android.preference.PreferenceFragment
import android.preference.PreferenceManager
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import project.android.headli9es.databinding.ActivityConfigBinding

class ConfigActivity : AppCompatActivity() {

    private lateinit var mBinding: ActivityConfigBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mBinding = DataBindingUtil.setContentView(this, R.layout.activity_config)

        mBinding.topAppBar.setNavigationOnClickListener { finish() }
    }

    class NewsConfigFragment : PreferenceFragment(), OnPreferenceChangeListener {
        override fun onCreate(savedInstanceState: Bundle?) {
            super.onCreate(savedInstanceState)
            addPreferencesFromResource(R.xml.configs)

            // TODO: Set the default news outlet to always load on open, regardless of selected,
            //  TEMPORARY outlet used at runtime.

            // Find the {@link ListPreference}s in configs.xml
            val newsOutlet = findPreference(getString(R.string.settings_news_outlet_key))
            val pageSize = findPreference(getString(R.string.settings_page_size_key))

            // Set and display summary of the {@link ListPreference} according to selected value
            setSummary(newsOutlet)
            setSummary(pageSize)
        }

        private fun setSummary(changedPreference: Preference) {
            changedPreference.onPreferenceChangeListener = this
            val preferences =
                PreferenceManager.getDefaultSharedPreferences(changedPreference.context)
            val preferenceString = preferences.getString(changedPreference.key, "")!!
            onPreferenceChange(changedPreference, preferenceString)
        }

        override fun onPreferenceChange(preference: Preference, newValue: Any): Boolean {
            val preferenceValue = newValue.toString()
            return if (preference is ListPreference) {
                val listPref = preference
                val indexOfSelectedValue = listPref.findIndexOfValue(preferenceValue)
                if (indexOfSelectedValue >= 0) {
                    val prefLabels = listPref.entries
                    preference.setSummary(prefLabels[indexOfSelectedValue])
                }
                true
            } else false
        }
    }
}